generic-openai-model-provider/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/obot-platform/tools/openai-model-provider/proxy"
+)
+
+func main() {
+	isValidate := len(os.Args) > 1 && os.Args[1] == "validate"
+
+	baseURL := os.Getenv("OBOT_GENERIC_OPENAI_MODEL_PROVIDER_BASE_URL")
+	if baseURL == "" {
+		fmt.Fprintln(os.Stderr, "OBOT_GENERIC_OPENAI_MODEL_PROVIDER_BASE_URL environment variable not set")
+		fmt.Printf("{ \"error\": \"BaseURL is required\" }\n")
+		os.Exit(1)
+	}
+
+	u, err := url.Parse(strings.TrimRight(baseURL, "/"))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid baseURL %q: %v\n", baseURL, err)
+		fmt.Printf("{ \"error\": \"Invalid BaseURL: %v\" }\n", err)
+		os.Exit(1)
+	}
+
+	if u.Scheme == "" {
+		if u.Hostname() == "localhost" || u.Hostname() == "127.0.0.1" {
+			u.Scheme = "http"
+		} else {
+			u.Scheme = "https"
+		}
+	}
+
+	cfg := &proxy.Config{
+		APIKey:          os.Getenv("OBOT_GENERIC_OPENAI_MODEL_PROVIDER_API_KEY"), // optional, as e.g. Ollama doesn't require an API key
+		ListenPort:      os.Getenv("PORT"),
+		BaseURL:         u.String(),
+		RewriteModelsFn: proxy.RewriteAllModelsWithUsage("llm"),
+		Name:            "Generic OpenAI",
+	}
+
+	if err := cfg.Validate("/tools/generic-openai-model-provider/validate"); err != nil {
+		os.Exit(1)
+	}
+
+	if isValidate {
+		return
+	}
+
+	if err := proxy.Run(cfg); err != nil {
+		panic(err)
```

Comment:
nit: `panic` usually prints not pretty messages. Would probably be best to just `Printf` everything we need to say and then `os.Exit`.

Reply:
done